src/Shared/src/PropertyHelper/PropertyHelper.cs

```diff
@@ -348,8 +347,10 @@ public static PropertyHelper[] GetVisibleProperties(Type type)
 
             if (value != null)
             {
-                foreach (var helper in GetProperties(value.GetType()))
+                var properties = GetProperties(value.GetType());
+                for (var i = 0; i < properties.Length; i++)
```

Comment:
Same here. The C#-compiler does the same, so I'll keep `foreach`.

Reply:
The compiler (and JIT) does not produce the same code, but the impact on throughput is not significant enough. I reverted this change.

Reply:
:+1:

> The compiler (and JIT) does not produce the same code

Not relevant for this PR (now), but it should be at least semantical equal code that the C# compiler generates when emitting IL for that foreach loop over an array. Maybe there is a difference between Debug / Release, but at least for Release it should behave as stated.

BTW 1: I see the correct behavior (on a stripped down code-variant of this PR) -- checked with Release.
BTW 2: Perf-wise hat matters anyway is that the bound-checks are elided, and this will be the case.